pkg/resources/machine/machinedeployment.go

```diff
@@ -49,7 +49,7 @@ func Deployment(c *kubermaticv1.Cluster, nd *apiv1.NodeDeployment, dc *kubermati
 	} else {
 		// GenerateName can be set only if Name is empty to avoid confusing error:
 		// https://github.com/kubernetes/kubernetes/issues/32220
-		md.GenerateName = fmt.Sprintf("%s-worker-", c.Spec.HumanReadableName)
+		md.GenerateName = fmt.Sprintf("%s-worker-", c.Name)
```

Comment:
Isn't this going to update existing `MachineDeployments`? What will happen to them?

Reply:
Hi @embik . This is only for new Machine Deployments with empty name. Before this PR, it was using the cluster name to generate the MD name and, because of that, when the cluster name had not accepted characters (e.g. space), the auto generated name was invalid and it wasn't possible to create the new machine deployment.

Reply:
it's only autogenerated name with the KKP cluster name as a prefix The KKP human-readable name  should support an arbitrary name

Reply:
Gotcha, thanks for explaining! it seems this is only called when converting an incoming API call to a `MachineDeployment` spec.